docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,263 @@
+```
+
+## Documentation of data set contents
+
+### Column descriptions
+
+* `rep`: representative peptide id for the cluster. Generated by running MMSeqs2 `linclust` on the peptipedia database with an identity threshold of 30%.
+* `member`: cluster member peptide id.
+* `member_sequence`: cluster member peptide sequence.
+* `group_bioactivity`: bioactivity label that is the same between all peptides in a given cluster.
+* `all_bioactivities`: all predicted bioactivities for a given peptide.
+* `go_id`: gene ontology ID.
+* `pfam_id`: PFAM (protein family) ID.
+* `pfam_name`: name of PFAM protein family.
+* `pfam_type`: type of PFAM protein family.
+* `aliphatic_index`: relative volume occupied by aliphatic side chains (Alanine, Valine, Isoleucine, and Leucine).
+* `boman_index`: The potential interaction index proposed by Boman (2003) is an index computed by averaging the solubility values for all residues in a sequence. It can be used to give an overall estimate of the potential of a peptide to bind to membranes or other proteins. A value greater than 2.48 indicates that a protein has high binding potential.
+* `charge`: the theoretical net charge of a peptide sequence based on the Henderson-Hasselbach equation. Computed at a pH of 7 with the Lehninger pKscale.
+* `hydrophobicity`: average of the hydrophobicity values of each residue using one of the 39 scales from different sources.
+* `instability_index`: predicts the stability of a protein based on its dipeptide composition. A protein whose instability index is smaller than 40 is predicted as stable, a value above 40 predicts that the protein may be unstable.
+* `isoelectric_point`: the pH at which a particular molecule or surface carries no net electrical charge.
+* `molecular_weight`: the sum of the mass of each amino acid in the peptide sequence.
+* `pd1_residue_volume`: physical descriptor related to the peptide residue volume.
+* `pd2_hydrophilicity`: physical descriptor related to the peptide hydrophilicity.
+* `z1_lipophilicity`: Zscale quantifying the lipophilicity of the peptide.
+* `z2_steric_bulk_or_polarizability`: Zscale modeling steric properties like steric bulk and polarizability.
+* `z3_polarity_or_charge`: Zscale quantifying electronic properties like polarity and charge.
+* `z4_electronegativity_etc`: Zscale relating to electronegativity, heat of formation, electrophilicity and hardness.
+* `z5_electronegativity_etc`: Zscale relating to electronegativity, heat of formation, electrophilicity and hardness.
+* `length`: length in amino acids of the peptide.
+
+### Notes about the data set
+
+* We artificially generated similarity in peptides with specific bioactivity labels by clustering at 30% identity. The opposite of this should be done in real ML applications -- sequences that are similar in content should be removed so the same signal isn't repeated over and over again and so that there isn't pollution between the training and testing data sets.
```

Comment:
I get this and it makes sense, but I think we should be careful not to bring this up in the lessons, because it's likely to be confusing to people.

Reply:
kk i agree! i just wanted to add it so that _we_ know (and anyone who is a passerby on this repo) that this isn't a real dataset that is suddenly magic for peptide bioactivity prediction 😂